### DISCUSSION TITLE
How to detect session switching?

### DISCUSSION THREAD

**Discussion by DrSequence** [quickfix-j/quickfixj#439] [Q&A] [answered]
Hello everyone!
I'm trying to find a solution to how I can catch the moment of switching from the main connection to the spare one.
As an example:

`
...

[session]
...
SocketConnectHost=localhost
SocketConnectPort=8081
SocketConnectHost1=localhost
SocketConnectPort2=8082
...
`

when the connection is disconnected, I can't distinguish it from a simple disconnect.
I was thinking of getting access via:
`
...  
Session session = Session.lookupSession(sessionid);   
session.getResponder();
`
but it turns out to be null after disconnect or reconnect.
I've spent with debug a lot of time and I'd like to process IoSession or just attribute of this.
`
"QFJ_RESET_IO_CONNECTOR" eq true.
`

How can I detect switching?
I'm looking forward to any ideas.
Have a nice day!

**Comment by chrjohn** [comment, marked as answer]
What about this?
https://github.com/quickfix-j/quickfixj/blob/7e9ca8542697c3ab51e2e9350d90234a197cdd86/quickfixj-core/src/main/java/quickfix/mina/IoSessionResponder.java#L90

And check the address on every connect and compare to the last address that you stored?

**Reply by DrSequence** [reply]
Yep, thanks, this will probably solve my problem, but I think it would be a good idea to be able to log a reconnection event inside the library.

**Reply by chrjohn** [reply]
Hmm, could you be more specific what exactly you expect? I mean the address to which the Initiator connects is indeed logged on every attempt.

Example:
```
[2021-11-12 08:00:00,411] [INFO ] event   : MINA session created: local=/x.x.x.x:53314, class org.apache.mina.transport.socket.nio.NioSocketSession, remote=/yyy.yyy.yyy.yyy:13502
```

**Reply by DrSequence** [reply]
I've revised my problem and I agree that your answer solves my problem. Thanks!

**Reply by chrjohn** [reply]
Thanks. While thinking more about it I am asking myself if there could be a more straightforward way to get this information, i.e. the address to which the initiator connects. Maybe this information could be added to the `onConnect()` callback of the `SessionStateListener`.
